lib/srv/db/postgres/test.go

```diff
@@ -203,6 +204,9 @@ func (s *TestServer) handleStartup(client *pgproto3.Backend) error {
 	// simulates cloud provider IAM auth.
 	if s.cfg.AuthToken != "" {
 		if err := s.handlePasswordAuth(client); err != nil {
+			if err := client.Send(&pgproto3.ErrorResponse{Code: pgerrcode.InvalidPassword, Message: err.Error()}); err != nil {
```

Comment:
nit: Wondering if `InvalidPassword` should be send only on IsAccessDeniedError  errors.
```go
   if trace.IsAccessDeniedError(err) {
	  client.Send(&pgproto3.ErrorResponse{Code: pgerrcode.InvalidPassword ...
  } 
  ```